tests/unit/store/settings.spec.js

```diff
@@ -18,20 +18,20 @@ describe("store/settings", () => {
   });
   test("When initialized, Then the customer_account_ids is an empty with no value assigned", () => {
     const array_of_customerids = store.state.settings.customer_account_ids;
-    expect(array_of_customerids.length).toEqual(0);
+    expect(array_of_customerids).toHaveLength(0);
   });
-  test("When initialized, Then the customer_account_ids is an empty with no value assigned", () => {
+  test("When initialized, Then the customer_account_ids when accessed via settings_store_module is an empty with no value assigned", () => {
     const array_of_customerids = settings_store_module.state()
       .customer_account_ids;
-    expect(array_of_customerids.length).toEqual(0);
+    expect(array_of_customerids).toHaveLength(0);
   });
   test("When initialized, Then the file_count and max_file_count is zero 0 as with no value assigned", () => {
     const value = store.state.settings.file_count;
     const max = store.state.settings.max_file_count;
-    expect(value).toEqual(0);
-    expect(max).toEqual(0);
+    expect(value).toStrictEqual(0);
+    expect(max).toStrictEqual(0);
   });
-  test("Store a single user detail can be assigned to the user_ids ", () => {
+  test("When initialized user_details of UUID, Nickname is empty, Then commit user_details which have valid user_id UUID and nickname assert the values in Vuex for user_id, UUID and nickname", () => {
```

Comment:
The fact that the values are null initially is a pre-condition of the test (to make sure that the values are different than after the commit), so that should be in the Given section, not When.  "Then" is all about assertions, so it shouldn't have any "commits" taking place.

So something like `Given the values of the user details are initially null, When a new array of user ids with 1 user is committed, Then the value of user_details is updated in the Vuex store.`
This could be better described

Reply:
Would like to retain the present title, its more informative.

Reply:
This isn't a question about something being more informative or not.  The current title is inaccurate---committing something cannot happen in the "Then" portion of a test.  The "Then"/"Assert" section of a test is all about what you are checking happened as a result of the aspect of the code you are testing.